### DISCUSSION TITLE
deploy kontejneru na RPi

### DISCUSSION THREAD

**Discussion by sartoricz** [ocervinka/plccoms-mqtt-bridge#16] [Q&A]
Zdravím, 

zkouším rozjet plccoms-mqtt-bridge na RPi4 (Debian Bookworm)  a dostanu chybu:
`root@rpi0:~# docker run --network host -v $(pwd)/deploy/docker/config.yaml:/etc/plccoms-mqtt-bridge/config.yaml ocervinka/plccoms-mqtt-bridge
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
exec /usr/local/openjdk-11/bin/java: exec format error
`
Je to tím RPi, nebo je zakopaný pes jinde?

Díky za hint.

**Comment by AcMetelka** [comment]
Pro RPi4 jsem si musel překlad udělat lokálně, už nepamatuju přesně, ale zřejmě přes docker-compose -build. Image pro tuto architekturu na  dockerhubu není. Plus nějaké malé úpravy v compose souboru. ¨

Tu mám nějaké poznámky z kompilace, ale ne vše musí stále platit a není na 100%, že to je vše, to jsem musel nastavit/upravit:

Netbeans tweaky pro plcmqttbridge aplikaci
	- Pridani depencies do jarka (zdroj https://discuss.gradle.org/t/how-to-include-dependencies-in-jar/19571): 
	jar {
	    manifest {
	        attributes 'Main-Class': 'plcmqttbridge.PlcMqttBridge'
	        attributes 'Multi-Release': 'true'
	    }
	    from {
	        configurations.compileClasspath.filter{ it.exists() }.collect { it.isDirectory() ? it : zipTree(it) }
	    }
	}
	- Chybka pro log4j2:  vyznaceno vyse zeleně >> "attributes 'Multi-Release': 'true'"
Chyba pri spusteni z duvodu podepsane knihovny: zip -d acmetelka.plcmqttbridge.jar 'META-INF/.SF' 'META-INF/.RSA' 'META-INF/*SF'   … odstrani ze zipu podepisovaci doplnky ( z https://vispud.blogspot.com/2019/03/exception-in-thread-main.html)

**Reply by sartoricz** [reply]
Díky za komentář. S Javou a překladem projektu nejsem úplně kamarád, tak jsem se projektem inspiroval a sepsal si něco podobného v perlu.  Hezké to není, ale běží to :)
